sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/ScannerFileServiceTest.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * SonarSource :: .NET :: Shared library
+ * Copyright (C) 2014-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.dotnet.tests;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Optional;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.sonar.api.batch.fs.FilePredicate;
+import org.sonar.api.batch.fs.FilePredicates;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.utils.log.LogTester;
+import org.sonar.api.utils.log.LoggerLevel;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ScannerFileServiceTest {
+  @Rule
+  public LogTester logTester = new LogTester();
+
+  @Test
+  public void isSupportedAbsolute_passes_correct_argument() {
+    // arrange
+    FileSystem fs = mock(FileSystem.class);
+    FilePredicates filePredicates = mock(FilePredicates.class);
+
+    ArgumentCaptor<String> argumentCaptor = ArgumentCaptor.forClass(String.class);
+    when(filePredicates.hasAbsolutePath(argumentCaptor.capture())).thenReturn(mock(FilePredicate.class));
+    when(fs.predicates()).thenReturn(filePredicates);
+    when(fs.hasFiles(any())).thenReturn(true);
+
+    // act
+    ScannerFileService sut = new ScannerFileService("key", fs);
```

Comment:
Not sure what `sut` stands for. Renaming artifact?
```suggestion
    ScannerFileService sfs = new ScannerFileService("key", fs);
```

Reply:
system under test - Martin Fowler abbreviation (see [Mocks Aren't Stubs](https://martinfowler.com/articles/mocksArentStubs.html)). Costin is using it as well, and given it's industry standard-ish (because Fowler is a guru), I think it's fine to have it in the codebase.